tests/test_algorithm.py

```diff
@@ -313,6 +314,40 @@ def handle_data(context, data):
         with self.assertRaises(TypeError):
             algo.run()
 
+    @parameterized.expand([
+        (-1000, 'invalid_base'),
+        (0, 'invalid_base'),
+    ])
+    def test_invalid_capital_base(self, cap_base, name):
+        """
+        Test that the appropriate error is being raised and orders aren't
+        filled for algos with capital base <= 0
+        """
+        algo_text = """
+def initialize(context):
+    pass
+
+def handle_data(context, data):
+    order(sid(24), 1000)
+        """
+        sim_params = SimulationParameters(
+            start_session=pd.Timestamp("2006-01-03", tz='UTC'),
+            end_session=pd.Timestamp("2006-01-06", tz='UTC'),
+            capital_base=cap_base,
+            data_frequency="minute",
+            trading_calendar=self.trading_calendar
+        )
+
+        with self.assertRaises(ZeroCapitalError) as exc:
+            # make_algo will trace to TradingAlgorithm,
+            # where the exception will be raised
+            algo = self.make_algo(script=algo_text, sim_params=sim_params)
+            algo.run()
```

Comment:
So since we're going to hit your error in the line above, this line is never being run, and probably shouldn't be included in the test. If the idea is to make flake8 not yell at you about unused assignment, you can just not assign for the `make_algo` call above.

Reply:
Yeah I added that to comply with flake8, just fixed it